### ISSUE TITLE
Retro Achievements passwords are subject to accidental shell parameter expansion

### ISSUE THREAD

**Issue by ianmacd** [dragoonDorise/EmuDeck#479]
My Retro Achievements password happens to contain the substring `$2`. EmuDeck expands this to a null string at installation time, causing an incorrect password to be saved to `retroarch.cfg`.

I've looked at the code and I can see that some effort is made in the helper functions to prevent this kind of unintended string interpolation, but for some reason it's not working in this case.

**Comment by Godsbane** [comment]
Thanks will test this character set out and see if we can stop this from happening.

**Comment by AngelofWoe** [comment]
This is a bug in the AppImage code. Parameters echoed in the ipcChannel need to be enclosed in quotes to prevent parameter expansions.

```sh

//Achievements
ipcChannel.sendMessage('bash-nolog', [
  `echo ${state.achievements.user} > $HOME/.config/EmuDeck/.rau`,
]);

ipcChannel.sendMessage('bash-nolog', [
  `echo ${state.achievements.pass} > $HOME/.config/EmuDeck/.rap`,
]);
```

Would be

```sh

//Achievements
ipcChannel.sendMessage('bash-nolog', [
  `echo '${state.achievements.user}' > $HOME/.config/EmuDeck/.rau`,
]);

ipcChannel.sendMessage('bash-nolog', [
  `echo '${state.achievements.pass}' > $HOME/.config/EmuDeck/.rap`,
]);
```

**Comment by dragoonDorise** [comment]
This should be fixed in 2.0.8. Please reopen this if you still find issues!
